.github/workflows/release.yml

```diff
@@ -0,0 +1,40 @@
+name: Release to npm
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'The version number to tag and release'
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    environment: release
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
```

Comment:
I would split it in 2 jobs:

1. publish the package to npm
2. create the GH release

in this way, if the GH release fails we know what to do, otherwise we must check if the npm release has been completed and then act accodingly.

In this way the `permissions` section should be:
- `contents:read` for the 1nd step ([ref](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings))
- as it is for the 2nd step instead